sympy/sets/tests/test_fancysets.py

```diff
@@ -802,3 +811,31 @@ def test_issue_16871b():
 def test_no_mod_on_imaginary():
     assert imageset(Lambda(x, 2*x + 3*I), S.Integers
         ) == ImageSet(Lambda(x, 2*x + I), S.Integers)
+
+
+def test_Rationals():
+    assert S.Integers.is_subset(S.Rationals)
+    assert S.Naturals.is_subset(S.Rationals)
+    assert S.Naturals0.is_subset(S.Rationals)
+    assert S.Rationals.is_subset(S.Reals)
+    assert S.Rationals.inf == -oo
+    assert S.Rationals.sup == oo
+    it = iter(S.Rationals)
+    assert [next(it) for i in range(12)] == [
+        0, 1, -1, S(1)/2, 2, -S(1)/2, -2,
+        S(1)/3, 3, -S(1)/3, -3, S(2)/3]
+    assert Basic() not in S.Rationals
+    assert S.Half in S.Rationals
+    assert 1.0 not in S.Rationals
+    assert 2 in S.Rationals
+    r = symbols('r', rational=True)
+    assert r in S.Rationals
+    raises(ValueError, lambda: x in S.Rationals)
```

Comment:
I get SympifyError here:
```
>>> from sympy import S, Symbol
>>> x = Symbol('x')
>>> x in S.Rationals
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/enojb/current/sympy/sympy/sympy/sets/sets.py", line 561, in __contains__
    symb = sympify(self.contains(other))
  File "/Users/enojb/current/sympy/sympy/sympy/sets/sets.py", line 288, in contains
    ret = sympify(self._contains(other))
  File "/Users/enojb/current/sympy/sympy/sympy/sets/fancysets.py", line 45, in _contains
    return _sympify(other.is_rational)
  File "/Users/enojb/current/sympy/sympy/sympy/core/sympify.py", line 417, in _sympify
    return sympify(a, strict=True)
  File "/Users/enojb/current/sympy/sympy/sympy/core/sympify.py", line 279, in sympify
    raise SympifyError(a)
sympy.core.sympify.SympifyError: SympifyError: None
```
Is that an intentional exception?

That also happens with `S.Rationals.contains(x)`. Does contains have to return True/False or is None a possibility there?

Reply:
The whole sets module needs a good audit for what must/can be returned from each method. Like should x in set raise an error if it is indeterminate? What about set.contains(x)? It seems like the latter should allow Contains as a result. I am not sure of the former.

Reply:
We can rule out some possibilities since `x in y` has to be a bool. Demo:
```python
class NotSure:
    def __bool__(self):
        return self

class Equivocal:
    def __contains__(self, other):
        return NotSure()

x in Equivocal()
```
Result:
```
TypeError: __bool__ should return bool, returned NotSure
```
So `x in y` only has 3 possibilities: True, False or exception.